src/Admin/ReaderThemes.php

```diff
@@ -210,17 +209,21 @@ public function get_default_reader_themes() {
 				]
 			);
 
-			if ( ! is_wp_error( $response ) ) {
-				set_transient( $cache_key, $response, DAY_IN_SECONDS );
+			if ( is_array( $response ) ) {
+				$response = (object) $response;
 			}
-		}
-
-		if ( is_wp_error( $response ) ) {
-			return [];
-		}
 
-		if ( is_array( $response ) ) {
-			$response = (object) $response;
+			/**
+			 * The response must minimally be an object with a themes array.
+			 *
+			 * @see https://wordpress.org/support/topic/issue-during-activating-the-updated-plugins/#post-13383737
+			 */
+			if ( is_wp_error( $response ) || ! is_object( $response ) || ! is_array( $response->themes ) ) {
```

Comment:
You would need to check if the `themes` property exists as well before accessing it, otherwise a `Undefined property` notice would occur.

```suggestion
			if (
				is_wp_error( $response )
				|| ! is_object( $response )
				|| ! property_exists( $response, 'themes' )
				|| ! is_array( $response->themes )
			) {
```